### ISSUE TITLE
Reloading window breaks fs.readdir, causing it to hang

### ISSUE THREAD

**Issue by darsain** [electron/electron#24338]
### Preflight Checklist

* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:** 9.0.5
* **Operating System:** Windows 10
* **Last Known Working Electron version:** I didn't work with electron for a long time, but last time I played with it the app had similar runtime reloading stuff going on and I don't remember it ever causing issues. That was Electron 1.6.2

### Expected Behavior

Reloading window with any of (renderer process) `location.reload()`, or (main process) `window.reload()`, `window.webContents.reload()`, `window.webContents.reloadIgnoringCache()`, should reload the process and work just as initial load of the same file.

### Actual Behavior

Whenever I try reloading window (no matter the method, as described above), my app initialization process hangs as soon as it touches `fs.readdir()` call, despite it working just fine on initial load. No errors or any indication or clue anywhere as to what might be happening.

Here is the snippet from the isolated example below:

```js
const FS = require('fs');
const {promisify} = require('util');
const readdir = promisify(FS.readdir);

async function load() {
	console.log('load(): start');
	try {
		await readdir('.'); // <= it hangs here on reload
		document.body.innerHTML = 'success! press F5 to reload';
	} catch (error) {
		document.body.innerHTML = `<pre>${error.message || error.stack}</pre>`;
	}
	console.log('load(): end');
}

load().then(console.log, console.error);
```

### To Reproduce

```sh
git clone https://gist.github.com/aa794ac2dc3662fee5d800da420a4813.git reloadbug
cd reloadbug
npm install
npm start
```

Hit F5 until it happens. On my app it happens almost 100% of the time. In the reproduction example around 50% of the time.

### Screenshots

![test](https://user-images.githubusercontent.com/538801/85945881-e6d53780-b940-11ea-83c9-92e81aa80efd.gif)

**Comment by marc2332** [comment]
Hello, I am afraid I am facing a similar/maybe the same issue.

I cannot read any file's content, example: 

```js
const fs = require("fs")

fs.readFile("C:/Users/mespi/Projects/Graviton-App/package.json",'UTF-8',(a,b)=>console.log(a,b))
```

At the same moment the count reaches 1 I reload the window and you can see it's content is printed very quickly ( the red text ). I made a GIF:

![Graviton-Editor-2020-07-02-21-38-56](https://user-images.githubusercontent.com/38158676/86402793-b547cd80-bcac-11ea-9a03-6503c298eae6.gif)

**Comment by marc2332** [comment]
I have the same specs as @darsain , Windows 10 and electron v9.0.5. I can ensure this was working on Electron v8.3.0.

**Comment by codebytere** [comment]
Is this failing for either of you on macOS? I just tried to reproduce there and could not

**Comment by marc2332** [comment]
> Is this failing for either of you on macOS? I just tried to reproduce there and could not

Sorry, I am a Windows 10 user but I will try it on Ubuntu as soon I can

**Comment by marc2332** [comment]
Just tried reading a file in Ubuntu and it **works**. I made the same test as in my GIF

Looks like a Windows-only bug I think

**Comment by marc2332** [comment]
Hello, I still have this issue in v10.

https://github.com/electron/electron/issues/22119 is totally related to my issue

```
An interesting observation is: when you reload again, for a brief moment before the page is refreshed, the previous fs calls are magically unblocked. This can be observed by console.log lines suddenly printing the moment you hit Ctrl+R which disappears quickly as the whole page reloads.
```

**Comment by codebytere** [comment]
Duplicate of https://github.com/electron/electron/issues/22119
